exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -1010,7 +1010,9 @@ public void addToEventQueue(final QueryState newState, final Exception exception
 
   private void recordNewState(final QueryState newState) {
     state = newState;
-    queryManager.updateEphemeralState(newState);
+    if (queryContext.getOptions().getOption(ExecConstants.ZK_QUERY_STATE_UPDATE)) {
+      queryManager.updateEphemeralState(newState);
+    }
```

Comment:
For long running queries, it may not make much difference. It adds latency of around ~50-60 msec for single query. However, with high concurrency, impact of contention because of zookeeper updates is significant. Like I mentioned in the JIRA, for concurrency=100, the average query response time for simple queries is 8 sec vs 0.2 sec with these updates disabled.  It does not impact the query profile. Query profile gets updated and written at the end of the query as usual.  This option affects only running queries. In Web UI, you will not see running queries and their state.